docs/docs/reference/providers/bash.md

```diff
@@ -98,20 +98,24 @@ The list of directories to cache between runs
 
 ### Ports
 
-If you'd like your workflow to expose ports, you have to specify the `ports` property with the number
-of ports to expose. Actual ports will be assigned on startup and passed to the workflow via the environment
-variables `PORT_<number>`.
+If you'd like your workflow to expose ports, you have to specify the `ports` property with the list
+of ports to expose. You could specify a mapping `APP_PORT:LOCAL_PORT` or just `APP_PORT` — in this
+case dstack will choose available `LOCAL_PORT` for you.
+
+!!! info "NOTE:"
+    Ports range `2000–2999` is reserved for dstack needs. However, you could remap them to different `LOCAL_PORT`s.
```

Comment:
1. Do we show an error if the user chooses a port within `2000-2999`?
2. Is it difficult to use `10000-10999` (to reduce the chances of an overlap)?

Reply:
1. I don't see how we could reliably differentiate remapping and port requests. For example, I want to run code on a different port: `-p 2001:3001`
2. I believe we can do that